packages/compartment-mapper/src/node-powers.js

```diff
@@ -1,5 +1,7 @@
 // @ts-check
 
+import './types.js';
```

Comment:
This is definitely one of those cases where I don't know how tsc + jsdoc types behaves. I have seen this pattern used and it somewhat baffles me that it works. I assume it's because jsdoc types end up creating ambient types in some cases?

It's reminiscent of mixed code and type imports that TypeScript started with, where you can sometimes end up with side effect import execution when you only need types. Nowadays a `.ts` file should have separate `import { foo } from './foo.js';` and `import type { Foo } from './foo.js';` so that type imports can be clearly removed when js is emited.

The way I would have done this is to have:
- the types written as a `.d.ts` file instead of `.js` with jsdoc (probably optional, but it would 100% be clear whether we end up with ambient types or not)
- The definition imported in jsdoc, either at the site of use, or as a local typedef: `/** @typedef {import('./types.js').ReadPowers} ReadPowers */`

Advantage is that you don't end up with potential runtime module cycles because of type definitions. Drawback is that type only files may not become part of the bundle output.

I don't know if there is an easy way to alias multiple types for local use (equivalent of `import type { ReadPowers, WritePowers } from './types.js';`

Reply:
I agree, this is uneasy footing. The defect was not detectable until after publishing the flaw.